commands/flags.go

```diff
@@ -0,0 +1,170 @@
+	return []cli.Flag{
+		&cli.StringSliceFlag{
+			Name:  FlagResourceType,
+			Usage: "Resource types to nuke. Include multiple times if more than one.",
+		},
+		&cli.StringSliceFlag{
+			Name:  FlagExcludeResourceType,
+			Usage: "Resource types to exclude from nuking. Include multiple times if more than one.",
+		},
+		&cli.BoolFlag{
+			Name:  FlagListResourceTypes,
+			Usage: "List available resource types",
+		},
+	}
+}
+
+// InspectResourceTypeFlags returns flags for resource type filtering in inspect commands
+func InspectResourceTypeFlags() []cli.Flag {
+	return []cli.Flag{
+		&cli.StringSliceFlag{
+			Name:  FlagResourceType,
+			Usage: "Resource types to inspect. Include multiple times if more than one.",
+		},
+		&cli.StringSliceFlag{
+			Name:  FlagExcludeResourceType,
+			Usage: "Resource types to exclude from inspection. Include multiple times if more than one.",
+		},
+		&cli.BoolFlag{
+			Name:  FlagListResourceTypes,
+			Usage: "List available resource types",
+		},
+	}
+}
+
+// CommonExecutionFlags returns flags for execution control
+func CommonExecutionFlags() []cli.Flag {
+	return []cli.Flag{
+		&cli.BoolFlag{
+			Name:  FlagDryRun,
+			Usage: "Dry run without taking any action.",
+		},
+		&cli.BoolFlag{
+			Name:  FlagForce,
+			Usage: "Skip nuke confirmation prompt. WARNING: this will automatically delete all targeted resources without any confirmation.",
+		},
+		&cli.StringFlag{
+			Name:  FlagTimeout,
+			Usage: "Resource execution timeout.",
+		},
+	}
+}
+
+// CommonOutputFlags returns flags for output formatting
+func CommonOutputFlags() []cli.Flag {
+	return []cli.Flag{
+		&cli.StringFlag{
+			Name:  FlagOutputFormat,
+			Usage: "Output format (table, json)",
+			Value: DefaultOutputFormat,
+		},
+		&cli.StringFlag{
+			Name:  FlagOutputFile,
+			Usage: "Write output to file instead of stdout (optional)",
+		},
+		&cli.StringFlag{
+			Name:    FlagLogLevel,
+			Value:   DefaultLogLevel,
+			Usage:   "Set log level",
+			EnvVars: []string{"LOG_LEVEL"},
+		},
+	}
+}
+
+// ConfigFlag returns the config file flag
+func ConfigFlag() cli.Flag {
+	return &cli.StringFlag{
+		Name:  FlagConfig,
+		Usage: "YAML file specifying matching rules.",
+	}
+}
+
+// RegionFlags returns region-related flags (applicable to both AWS and GCP)
+func RegionFlags() []cli.Flag {
+	return []cli.Flag{
+		&cli.StringSliceFlag{
+			Name:  FlagRegion,
+			Usage: "Regions to include. Include multiple times if more than one.",
+		},
+		&cli.StringSliceFlag{
+			Name:  FlagExcludeRegion,
+			Usage: "Regions to exclude. Include multiple times if more than one.",
+		},
+	}
+}
+
+// GCPProjectFlag returns the GCP project ID flag
+func GCPProjectFlag() cli.Flag {
+	return &cli.StringFlag{
+		Name:     FlagProjectID,
+		Usage:    "GCP Project ID to nuke resources from.",
+		Required: true,
+	}
+}
+
+// CombineFlags combines multiple flag slices into one
+func CombineFlags(flagSets ...[]cli.Flag) []cli.Flag {
+	var combined []cli.Flag
```

Comment:
Cibsuder to pre-allocate `combined` slice

Reply:
Good catch! In this case, CombineFlags is only called during CLI initialization (not in a hot path), and we're combining a small number of flag sets. The performance benefit of pre-allocation would be negligible here, while the two-pass approach adds complexity (e.g., calculating the length and preallocating). I think the simpler, more readable version is better suited for this use case. WDYT?